### PR TITLE
[WIP] (#6579) move observability with kiali istio.io test to docs repo

### PR DESCRIPTION
**Pull request by su225**
for #6579

### CODE REVIEW

File: tests/observability/scripts/perform_kiali_test.txt [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  if [ -z $KIALI_PORT_FORWARD_PID ]; then
+    echo 'It seems that Kiali port-forward PID is not set up properly'
+    kill $(netstat -ntpl | grep '127.0.0.1:8888' | awk '{ print $7 }' | cut -d'/' -f1)
+  else
+    kill $KIALI_PORT_FORWARD_PID
+  fi
+  exit $exit_code
+}
+
+## Setup port-forwarding for Kiali and register trap
```
Comment (elysdir) [inline tests/observability/scripts/perform_kiali_test.txt]: Change "Setup" to "Set up".  (https://developers.google.com/style/word-list#setup)

File: tests/observability/scripts/perform_kiali_test.txt [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+## for cleaning up on exit
+echo "Setting up port forwarding from 8888 to Kiali's API port"
+setup_kiali_port_forwarding 8888 # TODO: Pick a random unused port
+
+echo "Register clean up task for Kiali port-forward on exit"
```
Comment (elysdir) [inline tests/observability/scripts/perform_kiali_test.txt]:
```suggestion
echo "Register cleanup task for Kiali port-forward on exit"
```

File: tests/observability/scripts/perform_kiali_test.txt [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+trap cleanup_kiali_port_forwarding EXIT
+
+# Allow port-forwarding to be setup in the background
```
Comment (elysdir) [inline tests/observability/scripts/perform_kiali_test.txt]:
```suggestion
# Allow port-forwarding to be set up in the background
```

File: tests/observability/scripts/perform_kiali_test.txt [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+sleep 4
+
+
+## Login to Kiali using demo secret and save the token
```
Comment (elysdir) [inline tests/observability/scripts/perform_kiali_test.txt]:
```suggestion
## Log in to Kiali using demo secret and save the token
```

File: tests/observability/scripts/perform_kiali_test.txt [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+DEMO_SECRET='YWRtaW46YWRtaW4='
+API_TOKEN=$(curl -H "Authorization: Basic $DEMO_SECRET" $KIALI_LOCAL_URL/authenticate | jq -r '.token')
+if [ -z $API_TOKEN ]; then
+  echo 'Failed to obtain auth token for Kiali'
+  exit 1
+fi
+
+
+# Attaches JWT obtained from Kiali and send
+# API request to Kiali
```
Comment (elysdir) [inline tests/observability/scripts/perform_kiali_test.txt]: I think that "Attaches" should be "Attach". Or if not, then I think that "send" should be "sends".

File: tests/observability/scripts/perform_kiali_test.txt [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+function curl_with_kiali_cred() {
+  local api_path=$1
+  curl -H "Authorization: Bearer $API_TOKEN" $KIALI_LOCAL_URL/$api_path
+}
+
+## Determine Kiali's state
+echo "Checking Kiali state. Make sure it is running"
+KIALI_STATE=$(curl_with_kiali_cred status | jq -r '.status."Kiali state"') 
+if [ $KIALI_STATE != "running" ]; then 
+  echo "kiali is not running. So no point in proceeding further"
```
Comment (elysdir) [inline tests/observability/scripts/perform_kiali_test.txt]: Should "kiali" be capitalized here?

File: tests/observability/scripts/perform_kiali_test.txt [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+# Attaches JWT obtained from Kiali and send
+# API request to Kiali 
+function curl_with_kiali_cred() {
+  local api_path=$1
+  curl -H "Authorization: Bearer $API_TOKEN" $KIALI_LOCAL_URL/$api_path
+}
+
+## Determine Kiali's state
+echo "Checking Kiali state. Make sure it is running"
```
Comment (elysdir) [inline tests/observability/scripts/perform_kiali_test.txt]: I can't tell whether this means that the code is making sure that Kiali is running, or whether it's an instruction to the human reader to make sure that Kiali is running. Can you rephrase a little bit to clarify?

File: tests/observability/scripts/perform_kiali_test.txt [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+  echo "expected: $EXPECTED_WORKLOADS_STR"
+  echo "  actual: $ACTUAL_WORKLOADS_STR"
+  exit 1
+fi
+
+## STEP 2:
+## Obtain App graph which aggregates all versions of an app
+## to a single node.
+
+
```
Comment (elysdir) [inline tests/observability/scripts/perform_kiali_test.txt]: Are these steps intentionally left blank?

Comment (su225) [reply tests/observability/scripts/perform_kiali_test.txt]: Yes, Each of them involves parsing JSON. At that that I was not even sure what steps should be there. Since much of it would be tested by Kiali team anyways we would test integration with Istio.

**Comment by frankbu** [comment]
Closing this PR as it is obsolete.

@su225 The istio.io testing framework has been completely replaced with a [new simpler design](https://github.com/istio/istio.io/blob/master/tests/README.md). Some of the functions you wrote in [perform_kiali_test.txt](https://github.com/istio/istio.io/blob/05279adbee8b7852efb5a4004a0899012faff790/tests/observability/scripts/perform_kiali_test.txt) might still be useful in a new test, but the rest of this PR is not needed. If you'd like to try again with the new framework, please open a new PR.
